internal/provider/resource_gitlab_project_test.go

```diff
@@ -525,66 +616,48 @@ func TestAccGitlabProject_MergeTrains(t *testing.T) {
 	})
 }
 
-func TestAccGitlabProject_willError(t *testing.T) {
-	var received, defaults gitlab.Project
+func TestAccGitlabProject_willErrorOnAPIFailure(t *testing.T) {
+	var received gitlab.Project
 	rInt := acctest.RandInt()
-	defaults = gitlab.Project{
-		Namespace:                        &gitlab.ProjectNamespace{ID: 0},
-		Name:                             fmt.Sprintf("foo-%d", rInt),
-		Path:                             fmt.Sprintf("foo.%d", rInt),
-		Description:                      "Terraform acceptance tests",
-		TagList:                          []string{"tag1"},
-		RequestAccessEnabled:             true,
-		IssuesEnabled:                    true,
-		MergeRequestsEnabled:             true,
-		JobsEnabled:                      true,
-		ApprovalsBeforeMerge:             0,
-		WikiEnabled:                      true,
-		SnippetsEnabled:                  true,
-		ContainerRegistryEnabled:         true,
-		LFSEnabled:                       true,
-		SharedRunnersEnabled:             true,
-		Visibility:                       gitlab.PublicVisibility,
-		MergeMethod:                      gitlab.FastForwardMerge,
-		PrintingMergeRequestLinkEnabled:  true,
-		OnlyAllowMergeIfPipelineSucceeds: true,
-		OnlyAllowMergeIfAllDiscussionsAreResolved: true,
-		SquashOption:               gitlab.SquashOptionDefaultOff,
-		PackagesEnabled:            true,
-		PagesAccessLevel:           gitlab.PublicAccessControl,
-		BuildCoverageRegex:         "foo",
-		CIConfigPath:               ".gitlab-ci.yml@mynamespace/myproject",
-		CIForwardDeploymentEnabled: true,
-	}
-	willError := defaults
-	willError.TagList = []string{"notatag"}
+
 	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t) },
 		ProviderFactories: providerFactories,
 		CheckDestroy:      testAccCheckGitlabProjectDestroy,
 		Steps: []resource.TestStep{
 			// Step0 Create a project
 			{
-				Config: testAccGitlabProjectConfig(rInt),
+				Config: fmt.Sprintf(`
+					resource "gitlab_project" "this" {
+						name = "foo-%d"
+
+						visibility_level = "public"
+					}`, rInt),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckGitlabProjectExists("gitlab_project.foo", &received),
-					testAccCheckAggregateGitlabProject(&defaults, &received),
+					testAccCheckGitlabProjectExists("gitlab_project.this", &received),
 				),
 			},
 			// Step1 Verify that passing bad values will fail.
 			{
-				Config:      testAccGitlabProjectConfig(rInt),
-				ExpectError: regexp.MustCompile(`\stags\sexpected\s.+notatag.+\sreceived`),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckAggregateGitlabProject(&willError, &received),
-				),
+				Config: fmt.Sprintf(`
+					resource "gitlab_project" "this" {
+						name = "foo-%d"
+						repository_storage = "non-existing"
+
+						visibility_level = "public"
+					}`, rInt),
+				ExpectError: regexp.MustCompile(`repository_storage_moves: \[is invalid\]`),
```

Comment:
Where does the string `repository_storage_moves` come from?

Reply:
I asked that myself - it's what is returned by the upstream API ... but I guess it's related to the `repository_storage` stuff which this test does intentionally wrong ...

Reply:
I think this is worth a comment, or, changing the regex to just be "is invalid", just because `repository_storage_moves` is kind of foreign to the provider code and not really relevant to the asserted behavior

Reply:
done